modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -1288,9 +1302,38 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
         TypedNode.CollectionTN(collectionType, List.empty)
       case (node, Type.Map(_, _, _, _)) if node == Node.nullNode =>
         TypedNode.MapTN(List.empty)
+      case (node, IdRefCase()) =>
+        val ref = Type.Ref("smithy4s", "ShapeId")
+        val namespace :: name :: _ =
+          node.asStringNode.get.getValue.split("#").toList
+        def toField(value: String) = TypedNode.FieldTN.RequiredTN(
+          NodeAndType(
+            Node.from(value),
+            Type.PrimitiveType(Primitive.String)
+          )
+        )
+        TypedNode.StructureTN(
+          ref,
+          List("namespace" -> toField(namespace), "name" -> toField(name))
+        )
       case (node, tpe) => throw UnhandledTraitBinding(node, tpe)
     }
 
+  private object IdRefCase {
+    def unapply(tpe: Type): Boolean = tpe match {
+      case Type.ExternalType(
+            _,
+            fqn,
+            _,
+            _,
+            Type.PrimitiveType(Primitive.String),
+            _
+          ) if fqn == "smithy4s.ShapeId" =>
```

Comment:
suggestion: `==` => `===`, at least it forces you to use the same type on both sides